### ISSUE TITLE
A subscriber, once disconnected from an ExclusivePublication, won't automatically rejoin a new ExclusivePublication with the same sessionId

### ISSUE THREAD

**Issue by kinghchan** [real-logic/aeron#1460]
Say I start an ExclusivePublication using this channel: "aeron:udp?endpoint=localhost:3333|session-id", and I start a Subscriber on this exact same channel and wait for them to become connected.

Now say I close the ExclusivePublication - the Subscriber will go into a 'Not Connected' state.

Now say I reopen a new ExclusivePublication on the same channel - I would expect the Subscriber to automatically rejoin this channel and become 'Connected'. However, the Subscriber remains 'Not Connected'.

However, if my subscriber is SPYING on this channel, then it DOES automatically rejoin.

I was wondering if this is expected behaviour?

Thank you!

**Comment by mjpt777** [comment]
Why are you specifying the session-id?

**Comment by mjpt777** [comment]
Have you tried giving the Subscription sufficient time to reconnect by waiting in a loop and sleeping?

**Comment by kinghchan** [comment]
> Have you tried giving the Subscription sufficient time to reconnect by waiting in a loop and sleeping?

Yes I have - it would never reconnect, despite waiting for minutes. In this minimal example, if the subscription does indeed reconnect, I would expect the `poll` to return something > 0 eventually, but it never does.


> Why are you specifying the session-id?

That's because we need to ensure that a stream can only ever have 1 single Publication. We use the channel, stream ID and session ID together as a unique identifier for a stream. By specifying the session ID, we can ensure that no other publication can ever publish on the same stream. If we didn't specify the session ID, another ExclusivePublication could start publishing on the same channel and stream ID, in which case a Subscriber on the same stream would now receive messages from 2 different ExclusivePublications. Whereas if we did specify a session ID, the media driver would prevent another ExclusivePublication from trying to publishing messages on the same stream.

**Comment by mjpt777** [comment]
Should be fixed with this commit. https://github.com/real-logic/aeron/commit/58e8582e48015abc747c334083cf6945af41b893

**Comment by kinghchan** [comment]
Thank you Martin!
